### PR TITLE
Allow combining of `-Skip` and `-SkipLast` parameters in `Select-Object` cmdlet.

### LINKED ISSUE #11752: Select-Object: Allow combining -Skip and -SkipLast
See also: #13769

# Summary of the new feature/enhancement

`Select-Object`'s `-Skip` and `-SkipLast` parameters are currently in separate parameter sets, which syntactically prevents using them together.

However, there seems to be no good _conceptual_ reason to prevent that, as it is conceivable that you may want to skip elements at _both_ the beginning and end:

```powershell
# WISHFUL THINKING
1..5 | Select-Object -Skip 1 -SkipLast 1
2
3
4
```

* `-Skip`  should be applied first.

* `-SkipLast` must remain mutually exclusive with `-First` and `-Last`(which themselves can be combined with `-Skip`).

If the suggestion in #11750 is implemented, the same combination of parameters should also be allowed with `Get-Content`.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
# PR Summary

Fixes #11752

Modified parameter set for `Skip` to include `SkipLastParameter` parameter set.

## PR Context

Allow combining of `-Skip` and `-SkipLast` when using `Select-Object` cmdlet.

e.g.

```powershell
1, 2, 3 | Select-Object -Skip 1 -SkipLast 1
2
```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by ArmaanMcleod** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

**Comment by ArmaanMcleod** [comment]
@microsoft-github-policy-service agree

**Review by iSazonov — APPROVED** [review]

**Review by kilasuit — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1 [hunk 1]
```diff
@@ -231,6 +231,19 @@ Describe "Select-Object DRT basic functionality" -Tags "CI" {
...
+        $results.Count | Should -Be 2
+        $results[0] | Should -BeExactly "1"
+        $results[1] | Should -BeExactly "2"
+    }
+
+    It "Select-Object with Skip and SkipLast should work" {
+        $results = "1", "2", "3" | Select-Object -Skip 1 -SkipLast 1
+        $results.Count | Should -Be 1
+        $results[0] | Should -BeExactly "2"
+    }
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Select-Object.Tests.ps1:245]:
```suggestion
    It "Select-Object with Skip and SkipLast should work" {
        $results = "1", "2", "3" | Select-Object -Skip 1 -SkipLast 1
        $results.Count | Should -Be 1
        $results[0] | Should -BeExactly "2"
    }
    
    It "Select-Object with Skip and SkipLast should work with Skip overlapping SkipLast" {
        $results = "1", "2" | Select-Object -Skip 2 -SkipLast 1
        $results. Count | Should -Be 0
    }
    
    It "Select-Object with Skip and SkipLast should work with skiplast overlapping skip" {
        $results = "1", "2" | Select-Object -Skip 1 -SkipLast 2
        $results. Count | Should -Be 0
    }
```

**Pull request merged**